src/transformStateWithClones.js

```diff
@@ -6,8 +6,32 @@
  *
  * @return {Object[]}
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+function transformStateWithClones(initialState, actions) {
+  const stateHistory = [];
+  let currentState = { ...initialState }; // Клонування початкового стану
+
+  actions.forEach((action) => {
+    if (action.type === 'addProperties') {
+      // Додаємо властивості з extraData
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      // Видаляємо вказані властивості
+      currentState = Object.keys(currentState).reduce((newState, key) => {
+        if (!action.keysToRemove.includes(key)) {
```

Comment:
Check if `action.keysToRemove` is defined and is an array. This will prevent potential runtime errors if `keysToRemove` is undefined or not an array.